browser/url_sanitizer/url_sanitizer_browsertest.cc

```diff
@@ -161,48 +162,51 @@ class URLSanitizerTestBase : public InProcessBrowserTest {
 
       constexpr const char kClickButton[] =
           R"js(
-        (function() {
-          const button = document.getElementById('$1');
-          button.click();
-        })();
-    )js";
+            (function() {
+              const button = document.getElementById('$1');
+              button.click();
+            })();
+        )js";
       const auto script =
           base::ReplaceStringPlaceholders(kClickButton, {name}, nullptr);
       web_contents->Focus();
       ASSERT_TRUE(content::ExecJs(web_contents, script));
 
       const std::string text_from_clipboard = WaitClipboard();
 
-      EXPECT_TRUE(base::StartsWith(text_from_clipboard, "https://youtu.be/"))
+      EXPECT_EQ(expected_text, text_from_clipboard)
           << name << " " << should_sanitize << " " << text_from_clipboard;
-
-      if (should_sanitize) {
-        EXPECT_EQ(std::string::npos, text_from_clipboard.find("si="))
-            << name << " " << should_sanitize << " " << text_from_clipboard;
-      } else {
-        EXPECT_NE(std::string::npos, text_from_clipboard.find("si="))
-            << name << " " << should_sanitize << " " << text_from_clipboard;
-      }
     };
 
     const bool should_sanitize =
         base::FeatureList::IsEnabled(features::kBraveCopyCleanLinkFromJs);
 
-    check("test_1", should_sanitize);
-    check("test_2", should_sanitize);
-    check("test_3", should_sanitize);
-    check("test_4", should_sanitize);
-    check("test_5", should_sanitize);
+    const std::string sanitized = "https://youtu.be/B";
+    const std::string unsanitized = "hTtPs://Youtu.Be/B?si=oLb865I64uJlLRJX";
+    const std::string& expected = (should_sanitize) ? sanitized : unsanitized;
+
+    check("test_1", should_sanitize, expected);
+    check("test_2", should_sanitize, expected);
+    check("test_3", should_sanitize, expected);
+    check("test_4", should_sanitize, expected);
+    check("test_5", should_sanitize, expected);
+    // We cannot distinguish the context, so even if a password similar to the
+    // URL is copied, we will sanitize it.
+    check("test_sanitizable_password", should_sanitize, expected);
+    // Not sanitazable password should be copied as is.
+    check("test_not_sanitizable_password", should_sanitize, "Pa$$w0rd");
 
     const GURL no_permission_url(
         "https://no_permission.com/url_sanitizer/js_api.html");
     ASSERT_TRUE(ui_test_utils::NavigateToURL(browser(), no_permission_url));
 
-    check("test_1", false);
-    check("test_2", false);
-    check("test_3", false);
-    check("test_4", false);
-    check("test_5", false);
+    check("test_1", false, unsanitized);
+    check("test_2", false, unsanitized);
+    check("test_3", false, unsanitized);
+    check("test_4", false, unsanitized);
+    check("test_5", false, unsanitized);
+    check("test_sanitizable_password", should_sanitize, unsanitized);
+    check("test_not_sanitizable_password", should_sanitize, "Pa$$w0rd");
```

Comment:
Could we also add these test cases:
- `A:^C,D`
- `Ftp://Example.Com/?si=12345`

Neither of them should be sanitized. One is the test case from the password manager bug. The other is scheme we don't support.